tests/Functional/SearchTest.php

```diff
@@ -681,6 +681,39 @@ public function testEqualFilter(string $filter, array $expectedHits): void
         ]);
     }
 
+    public function testEscapingFilterValues(): void
+    {
+        $configuration = Configuration::create()
+            ->withFilterableAttributes(['title'])
+        ;
+        $title = "The 17\" O'Conner && O`Series \n OR a || 1%2 book?";
```

Comment:
In redisearch i did stumble over some more issues so I extended this string with:

```suggestion
        $title = "^The 17\" O'Conner && O`Series \n OR a || 1%2 \n book? \r \twhat \\ text // ok? end$";
```